mina-rosetta/templates/_helpers.tpl

```diff
@@ -0,0 +1,62 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "mina-rosetta.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "mina-rosetta.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
```

Comment:
I came across couple of times now, having a .Chart.Name here would create very long names .Chart.Name-.Release.name. In order to keep names readable and not exceed 64character limit, in other charts we use:
```
{{- $name := default .Release.name .Values.nameOverride }}

Reply:
I had a thought about updating 
```
{{- $name := default .Chart.Name .Values.nameOverride }}
```
to
```
{{- $name := default .Release.Name .Values.nameOverride }}
```
and as it may make our life easier, I really don't think it is a good idea.

**- Consistency with Helm Conventions:**
The _helper.tpl file is generated by the `helm create` command, which aligns with the Helm conventions. Changing it to .Release.Name will deviate from these conventions, and if our intend is to share these chart with the community, this will potentially lead to confusion for Helm users who are accustomed to the default behavior.

**- Chart Portability:**
Helm charts are often shared and reused across different projects or environments. Using .Chart.Name provides a consistent and portable way to name applications within the chart. Overriding it with .Release.Name might make the chart less portable and more tied to the specific release names in each deployment.

**- Intention Clarity:**
The use of .Chart.Name in the original implementation explicitly states that the app name is derived from the chart name. Changing it to .Release.Name will lead to ambiguity regarding the source of the app name, making it less clear for the community for maintaining or deploying the Helm chart.

**- Compatibility Concerns:**
The change might introduce compatibility issues with existing deployments or automation scripts that expect the default behavior. We have multiple charts referring one anothers (e.g. service names). Changing this behaviour may need to be updated, potentially causing disruptions.

**- Code Redundency:**
This block `{{- $name := default .Release.Name .Values.nameOverride }}` introduce redundency from the fact that the same condition contains $name .Release.Name is now checked twice.
As a consequence, we would have to refactor this block to make it clean for ourselves and the community.

**Which problem are we trying to address?**

If the readability issue stems from the constraint that the name is limited to 63 characters and then truncated, we have two variables at our disposal to tailor our Helm Chart to meet our specific requirements:
- .nameOverride
- .fullnameOverride
By leveraging these variables, we can enhance the readability of our Helm Chart and ensure that the generated names align with our preferences and constraints.

Reply:
these are valid points. Using name override instead of steering away from default helm implementations in the long run indeed can backfire.
As to which problem we're trying to address - simple example, following failed to create postgresql subchart resources because of the 64char constraint. Not too mention the fact that `kubectl get pods` output became unreadable.
<----------chart name-------------><--------------------release name -----------------------------><-subchart->
delegation-verify-coordinator-staging-testworld2-delegation-verify-coordinator-postqresql-0

ofc to solve this we can use full name override and change how we name things all together, i.e. not using platform name on a platform where we're running it as it does not make a name more distinctive.

Reply:
>ofc to solve this we can use full name override and change how we name things all together, i.e. not using platform name on a platform where we're running it as it does not make a name more distinctive.

You are right, I completly agree with you
appending  platform-name and testnet-name to the pod name is not really relevant as it does not bring any value as we still have the ability to narrow down logs and monitoring in general to a specific namespace

Reply:
i quite like the idea of turning

```
NAME                                                            READY   STATUS    RESTARTS      AGE
production-testworld-2-0-mina-archive-686c9c76bd-zmw4s          2/2     Running   0             12m
production-testworld-2-0-mina-archive-postgresql-0              1/1     Running   0             46m
production-testworld-2-0-mina-bp-0-7cbc84879d-dx8rn             1/1     Running   2 (17m ago)   27m
production-testworld-2-0-mina-bp-0-7fd69cdf7d-z6nk7             1/2     Running   1 (52s ago)   7m55s
production-testworld-2-0-mina-seed-0-655666844d-xsr28           2/2     Running   0             8m10s
production-testworld-2-0-offline-mina-rosetta-cbb887c59-hsmpp   1/1     Running   0             27m
production-testworld-2-0-online-mina-rosetta-6f8b4c6d59-9pvw8   1/1     Running   0             27m
```
into this
```

NAME                                                            READY   STATUS    RESTARTS      AGE
mina-archive-686c9c76bd-zmw4s                                   2/2     Running   0             12m
mina-archive-postgresql-0                                       1/1     Running   0             46m
mina-block-producer-0-7fd69cdf7d-z6nk7                          2/2     Running   1 (52s ago)   7m55s
mina-seed-0-655666844d-xsr28                                    2/2     Running   0             8m10s
mina-rosetta-offline-cbb887c59-hsmpp                            1/1     Running   0             27m
mina-rosetta-online-6f8b4c6d59-9pvw8                            1/1     Running   0             27m
```